### ISSUE TITLE
[a11y] With maximum accessibility font size, action buttons in AlertDialogs cause rendering overflow exceptions

### ISSUE THREAD

**Issue by jkurtw** [flutter/flutter#42696]
b/142592672

1. On iOS, go to Settings -> General -> Accessibility -> Larger Text and set to maximum.

2. Observe dialogs with actions can have rendering exceptions.

Full repro code:
```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: Scaffold(
        appBar: AppBar(title: Text('Alert Dialog Test')),
        body: Builder(
          builder: (context) => Center(
            child: RaisedButton(
              child: Text('Show Alert'),
              onPressed: () {
                showDialog(
                  context: context,
                  builder: (context) {
                    return AlertDialog(
                      title: Text('Alert Dialog'),
                      content: Text('Alert text'),
                      actions: [
                        FlatButton(onPressed: () {}, child: Text('Decline')),
                        FlatButton(onPressed: () {}, child: Text('Accept')),
                      ],
                    );
                  },
                );
              },
            ),
          ),
        ),
      ),
    );
  }
}
```
(We would like the text to show up fully, and we we would like them to be stacked vertically if they do not fit).

**Comment by Hixie** [comment]
This has been fixed!

**Comment by shihaohong** [comment]
Since https://github.com/flutter/flutter/pull/43226 was reverted, part of the problem has come back and we will need to re-design a solution that is less of a breaking change.

**Comment by shihaohong** [comment]
Just posting an update here that I'm still actively working on this issue. It's been taking a long time due to unforeseen breaking changes and test failures in google3 that require fixes to the upstream framework before we can proceed.

**Comment by shihaohong** [comment]
The behavior in the repro code has already been fixed, so I'll now close the issue
